pkg/liqo-controller-manager/resource-request-controller/resource-monitors/external-monitor.go

```diff
@@ -50,49 +47,55 @@ func NewExternalMonitor(ctx context.Context, address string) (*ExternalResourceM
 
 // Register sets an update notifier.
 func (m *ExternalResourceMonitor) Register(ctx context.Context, notifier ResourceUpdateNotifier) {
-	stream, err := m.ResourceReaderClient.Subscribe(ctx, &SubscribeRequest{})
-	if err != nil {
-		klog.Errorf("grpc error while subscribing: %s", err)
-	}
+	var err error
+	var stream ResourceReader_SubscribeClient
 	go func() {
-		for {
-			_, err := stream.Recv()
-			if errors.Is(err, io.EOF) {
-				klog.V(4).Infof("The external monitor closed the Register() stream")
-				break
+		for ctx.Err() == nil {
+			if stream == nil {
+				if stream, err = m.Subscribe(ctx, &Empty{}); err != nil {
+					klog.Errorf("Failed to subscribe to the external resource monitor: %s", err)
+					// Prevent busy loop in case of continuous errors
+					timeout, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+					<-timeout.Done()
+					cancel()
+					continue
+				}
+				// A unidirectional stream is enough to get notified by the server
+				err := stream.CloseSend()
+				if err != nil {
+					klog.Warningf("error while making stream unidirectional to external resource monitor: %w", err)
+				}
 			}
+			notification, err := stream.Recv()
 			if err != nil {
-				klog.Errorf("grpc error while receiving notifications: %s", err)
+				klog.Errorf("stream to external resource monitor server closed due to an error: %w", err)
```

Comment:
```suggestion
				klog.Errorf("stream to external resource monitor server closed due to an error: %v", err)
```

Do not use `%w` when printing, only when wrapping an error.